app/models/state_machines/user_programme_enrolment_state_machine.rb

```diff
@@ -21,8 +21,12 @@ class StateMachines::UserProgrammeEnrolmentStateMachine
     enrolment.programme.mailer.with(user: enrolment.user).pending.deliver_later
   end
 
-  after_transition(to: :complete) do |programme_enrolment|
+  before_transition(to: :complete) do |programme_enrolment, transition|
     # Set the name that should display on the certificate
+    transition.metadata["certificate_name"] = programme_enrolment.programme.programme_complete_counter.get_next_number
```

Comment:
Is this meant to be `certificate_number`? It looks like we immediately override the `certificate_name` in `set_user_programme_enrolment_complete_data` and then try to pull out a `certificate_number` from the transition metadata in the certificate generator.

```suggestion
    transition.metadata["certificate_number"] = programme_enrolment.programme.programme_complete_counter.get_next_number
```

Reply:
You are correct, I have used the wrong one!